src/toast/ops/totalconvolve.py

```diff
@@ -568,6 +597,100 @@ def conv_and_interpol(self, skycomp, beamcomp, lmax, mmax, pnt, nthreads):
         plan.interpol(cube, 0, 0, pnt[:, 0], pnt[:, 1], pnt[:, 2], res)
         return res
 
+    # MPI version with shared memory tricks, storing the full data cube only
+    # once per node.
+    def conv_and_interpol_mpi_new(self, skycomp, beamcomp, lmax, mmax, pnt, nthreads):
+        from multiprocessing import shared_memory
+
+        # create a communicator "intercomm" with exactly one task per node
+        intracomm = self.comm.Split_type(MPI.COMM_TYPE_SHARED)
+        intercomm = self.comm.Split(intracomm.rank == 0)
+        if intracomm.rank == 0:  # we are on the master node of intracomm
+            plan = totalconvolve.ConvolverPlan(
+                lmax=lmax,
+                kmax=mmax,
+                sigma=self.oversampling_factor,
+                epsilon=self.epsilon,
+                nthreads=intracomm.size * nthreads,
+            )
+            myrank, nranks = intercomm.rank, intercomm.size
+
+            # get shared memory for the data cube
+            shm = shared_memory.SharedMemory(
```

Comment:
TOAST uses https://github.com/tskisner/pshmem for shared memory management on node. @tskisner do you know if there are critical differences between `pshmem` and `multiprocessing.shared_memory`?

I also noticed some warnings from `shared_memory` after running the unit tests:
```
/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 10 leaked shared_memory objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/resource_tracker.py:229: UserWarning: resource_tracker: '/psm_8558dd07': [Errno 2] No such file or directory: '/psm_8558dd07'
  warnings.warn('resource_tracker: %r: %s' % (name, e))
/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/resource_tracker.py:229: UserWarning: resource_tracker: '/psm_177d1d75': [Errno 2] No such file or directory: '/psm_177d1d75'
...
```
I wonder if they are cause for concern.

Reply:
I wasn't aware of psmem ... if it is not too hard for me to switch, I'll do that.

Concerning the warnings, I see them too, but only intermittently, so I have no idea what's going on. Given that every MPI test run ends with 
```
 ----------------------------------------------------------------------
Ran 102 tests in 109.361s

[1] FAILED (errors=1)
[0] FAILED (errors=1)
ICE default IO error handler doing an exit(), pid = 27215, errno = 32
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[62616,1],0]
  Exit code:    1
--------------------------------------------------------------------------
```
I wondered whether there is some more fundamental problem with cleaning up after the tests, so I didn't think about this too much.

Reply:
OK, I think with `pshmem` things become even simpler!
I'm only a bit worried because I'm deliberately not using `pshmem`'s auto-sync feature. I'm just using its `data` member as a numpy array and take care of synchronization across nodes myself. Is that proper usage, @tskisner?

Reply:
Using `pshmem` at least offers the advantage that you don't need to communicate the pieces of the shared memory between nodes. `pshmem` can maintain synchronized copy of the same buffer on all nodes.

Reply:
I think I won't be able to avoid explicit communication, since I pass sub-arrays of the shared memory region to other functions, and they write to them without knowing that they are special. Also, synchronizing every write access to the buffer would be very expensive, and it is not necessary. If it's at all possible, I'd like to stick to the few explicit `Bcast` statements I currently have.